### ISSUE TITLE
`$te` does not work for "dot.keys" notation

### ISSUE THREAD

**Issue by Erythros** [nuxt-modules/i18n#2684]
### Environment

------------------------------
- Operating System: Linux
- Node Version:     v18.18.0
- Nuxt Version:     3.9.1
- CLI Version:      3.10.0
- Nitro Version:    2.8.1
- Package Manager:  npm@9.4.2
- Builder:          -
- User Config:      devtools, modules, i18n
- Runtime Modules:  @nuxtjs/i18n@8.0.0
- Build Modules:    -
------------------------------

### Reproduction

https://stackblitz.com/edit/nuxt-starter-llkq3b?file=app.vue

### Describe the bug

When using one-word keys for translations, like `welcome`, everything works fine. However, having a key like `welcome.home` makes `$te` not work properly.

Check the reproduction for a short example.

### Additional context

```ts
  messages: {
    en: {
      welcome: 'Welcome',
      "welcome.home": "welcome home EN"
    },
    fr: {
      welcome: 'Bienvenue',
      "welcome.home": "welcome home FR"
    },
  }
  ```

`$te('welcome')` works correctly.
`$te('welcome.home'` is always false

**Comment by BobbieGoede** [comment]
I don't think this message format is supported, see https://github.com/intlify/vue-i18n-next/issues/271. It does work the other way around:

```ts
messages: {
  en: {
    welcome: {
      greeting: 'Welcome',
      home: "welcome home EN"
    },
  fr: {
    welcome: {
      greeting: 'Bienvenue',
      home: "welcome home FR"
  },
}
```

Usage: `$t('welcome.greeting')` or `$t('welcome.home')`

**Comment by Cry0nicS** [comment]
It depends if a "flat" list is used for translations or an object.
The `dot.key` notation would work for an object. E.g. 
```ts
welcome: {
  default: 'Welcome',
  home: 'Welcome home'
}
```

Otherwise, for a flat list like in the given example, `$te(['welcome.home'])` can be used.
